### DISCUSSION TITLE
Search does not return result for new items when search a field value

### DISCUSSION THREAD

**Discussion by AHTei** [microsoft-search/pnp-modern-search#3973] [Q&A]
We are using PnP-Modern-Search V4 for Site Pages. We have added metadata fields for "Article ID" as text along with other fields.

We have migrated contents from other environment to the SharePoint Site Pages and are able to search them successfully using Article ID values. However, when a new page (article) is created, and Article ID gets a new unique value along its name then searching for this article works fine using its name but not using its Article ID value. We already verified that the Article ID value is present in the related article data load contents.

Please advise what would be the potential issue and how we could resolve it.

**Comment by kasperbolarsen** [comment]
I would suggest using one of the tools from here, https://microsoft-search.github.io/pnp-modern-search/using-query-tools-to-verify-issues/   to verify the data being returned from the API.
If your data isn't in the data being returned, it might be because the content hasn't been indexed resently. I wrote about how you can verify that:
https://www.m365thinking.com/post/what-to-do-when-the-sharepoint-search-doesn-t-pick-up-your-content-part-1-of-4

If you discover that the content hasn't been indexed for some time, like 3 or 4 days, I recommend creating a support ticket with Microsoft, as the issue is with them
